torchtune/modules/optim/_cpu_offload.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from torchtune.config._utils import _get_component_from_path
+
+try:
+    from torchao.prototype.low_bit_optim import (
```

Comment:
Should we have the try/except only on the import, instead of the whole function? Then we can also simply this file

Reply:
So are you proposing that if there is CPU offload optimizer from torchao, `CPUOffloadOptimizer` in torchtune shouldn't be defined at all? If it is the case, the problem is that I won't be able to import it in `__init__.py` (or I need to add extra version check/try-except logic again). Maybe I don't need to re-import it in `__init__.py`?

Reply:
Good point! I didnt think about the init.

@joecummings, is there any case like this in torchtune?

We can do:
`from torchtune.modules.optim.cpu_offload_optimizer import CPUOffloadOptimizer`

or

`from torchtune.modules.optim import CPUOffloadOptimizer`

But in the second case, importing it in the __init__ requires the implementation of a DummyClass with NotImplementedError. Do you see anyway around it? Or should we just have the DummyClass?